merlin/models/tf/blocks/optimizer.py

```diff
@@ -0,0 +1,429 @@
+            self._track_trackable(optimizer, name=f"Optimizer{1+len_exist_optimizers}")
+        self.update_optimzier_and_block.append(optimizer_blocks)
+        return
+
+    def get_config(self):
+        config = dict()
+        config = tf_utils.maybe_serialize_keras_objects(self, config, ["default_optimizer"])
+        config["name"] = self.name
+        config["optimizers_and_blocks"] = []
+        for optimizer_blocks in self.optimizers_and_blocks:
+            config["optimizers_and_blocks"].append(optimizer_blocks.get_config())
+        for optimizer_blocks in self.update_optimzier_and_block:
+            config["update_optimizers_and_blocks"].append(optimizer_blocks.get_config())
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        config["default_optimizer"] = tf.keras.optimizers.deserialize(config["default_optimizer"])
+        optimizers_and_blocks, update_optimizers_and_blocks = [], []
+        for optimizer_blocks_config in config["optimizers_and_blocks"]:
+            optimizers_and_blocks.append(OptimizerBlocks.from_config(optimizer_blocks_config))
+        config.update({"optimizers_and_blocks": optimizers_and_blocks})
+        if config.get("update_optimizers_and_blocks"):
+            for optimizer_blocks_config in config["update_optimizers_and_blocks"]:
+                update_optimizers_and_blocks.append(
+                    OptimizerBlocks.from_config(optimizer_blocks_config)
+                )
+        config.update({"update_optimizers_and_blocks": update_optimizers_and_blocks})
+        return cls(**config)
+
+    @property
+    def iterations(self):
+        """See base class."""
+        # Returning iterations from the first optimizer.
+        return self.optimizers_and_blocks[0].optimizer.iterations
+
+    @iterations.setter
+    def iterations(self, variable):
+        """See base class."""
+        for optimizer_blocks in self.optimizers_and_blocks:
+            optimizer_blocks.optimizer.iterations = variable
+
+    def variables(self):
+        """Returns the optimizer's variables."""
+        # OptimizerV2.variables() returns self._weights, so override that method.
+        return self.weights
+
+    @property
+    def weights(self) -> List[tf.Variable]:
+        """Returns the optimizer's variables."""
+        weights = []
+        for optimizer_blocks in self.optimizers_and_blocks:
+            weights += optimizer_blocks.optimizer.weights
+        return weights
+
+    @property
+    def optimizers(self) -> List[tf.keras.optimizers.Optimizer]:
+        """Returns the optimizers in MultiOptimizer (in the original order). Note: default_optimizer
+        is included here"""
+        return [pair.optimizer for pair in self.optimizers_and_blocks] + [self.default_optimizer]
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class LazyAdam(tf.keras.optimizers.Adam):
+    """Variant of the Adam optimizer that handles sparse updates more efficiently.
```

Comment:
We could add a reference to the source code of tf_addons ([here](https://github.com/tensorflow/addons/blob/v0.17.0/tensorflow_addons/optimizers/lazy_adam.py#L30-L144))

Reply:
Fixed, Thank you Sara